### PR TITLE
feat:created registerForm

### PR DESCRIPTION
**Pull request by manusankar9**

### CODE REVIEW

**Comment by manusankar9** [comment]
hi @sivasankar9 ,
still working on api.

File: src/components/full-calender/components/shared/register-form/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { emailAvailabilityService, registerUserDetailsService, userAvailabilityService} from '../service';
+
+export const userAvailability = username => {
+	const payload = { username };
+
+	return async dispatch => {
+		const response = await userAvailabilityService(payload);
+
+		if (response) {
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/actions/index.js]: remove the check, it's not required.

File: src/components/full-calender/components/shared/register-form/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { emailAvailabilityService, registerUserDetailsService, userAvailabilityService} from '../service';
+
+export const userAvailability = username => {
+	const payload = { username };
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/actions/index.js]: remove the unecessary wrapping ofr username,

File: src/components/full-calender/components/shared/register-form/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			dispatch({
+				type: "EMAIL_AVAILABILITY_DATA",
+				payload: { ...response, value: email }
+			});
+		}
+	};
+};
+export const emailAvailabilityReset = payload => {
+	return dispatch => {
+		if (payload.email.length === 0) {
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/actions/index.js:41]: update to  use reg expression.

File: src/components/full-calender/components/shared/register-form/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			dispatch({
+				type: "RESET_EMAIL_DATA"
+			});
+		}
+	};
+};
+export const passwordAvailability = payload => {
+	return dispatch => {
+		const isError = !payload.length > 0;
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/actions/index.js]: please use utility  reg and rename the const **isError** to  _error_

File: src/components/full-calender/components/shared/register-form/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import { connect } from "react-redux";
+
+const mapStateToProps = state =>({
+    registerFormUsername: state.registerFormCalender.username,
+    registerFormEmail: state.registerFormCalender.email,
+    registerFormPassword: state.registerFormCalender.password
+    
+});
+
+export default connect(mapStateToProps, {registerUserDetails,passwordAvailabilityReset, userAvailability, userAvailabilityReset, emailAvailabilityReset, emailAvailability, passwordAvailability})(RegisterForm);
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/index.js]: create a const actions = { }  drop all the actions in it.

File: src/components/full-calender/components/shared/register-form/reducer/password.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const initialState = {
+    error: true,
+    value: ""
+};
+
+export default (state = initialState, action) => {
+    switch (action.type) {
+        case "PASSWORD_AVAILABILITY": return action.payload;
+        case "RESET_PASSWORD_DATA": return initialState;
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/reducer/password.js]: it should be ...
        case "RESET_PASSWORD_DATA":
        case "RESET_REGISTER_FORM":
                  return initialState;

File: src/components/full-calender/components/shared/register-form/reducer/usename.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const initialState = {
+    error: true,
+    value: "",
+    message: ""
+};
+
+export default (state = initialState, action)=>{
+    switch (action.type) {
+        case "USER_AVAILABILITY_DATA":return action.payload;
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/reducer/usename.js:9]: it should be ...
case "RESET_USERNAME_DATA":
case "RESET_REGISTER_FORM":
return initialState;

File: src/index.js [hunk 8]
```diff
@@ -5,13 +5,15 @@ import {Provider} from 'react-redux';
 import React from 'react';
 import calenderEvents from './components/full-calender/reducer';
```
Comment (sivasankar9) [inline src/index.js]: import should be **fullCalender**

File: src/index.js [hunk 9]
```diff
@@ -5,13 +5,15 @@ import {Provider} from 'react-redux';
...
 import loginCalender from './components/full-calender/components/shared/login/reducer';
+import registerFormCalender from './components/full-calender/components/shared/register-form/reducer';
 import { render } from 'react-dom';
 import thunkMiddleware from 'redux-thunk';
  
 
 let store = createStore(combineReducers({
     fullCalender: calenderEvents,
-    loginCalender: loginCalender
+    loginCalender: loginCalender,
```
Comment (sivasankar9) [inline src/index.js]: it should be only
 **loginCalender**,
**registerFormCalender**

File: src/components/full-calender/components/shared/register-form/component/index.js [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+    const {registerFormEmail: {error: emailError}, registerFormPassword: {error: passwordError}, registerFormUsername: {error: usernameError}} = props;
+
+    let flg = [!emailError, !passwordError, !usernameError].every(a => a);
+
+    return flg;
+  };
+
+  const userhandlerBlur = (e) => {
+
+    if (e.target.value !== "") {
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/component/index.js]: use util function.

File: src/components/full-calender/components/shared/register-form/component/index.js [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+  };
+  
+  const userhandlerChange = (e) => {
+    props.userAvailabilityReset({ username: e.target.value });
+
+  };
+
+  const emailhandlerBlur = (e) => {
+
+    if (e.target.value !== "") {
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/component/index.js]: use util function.

File: src/components/full-calender/components/shared/register-form/component/index.js [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+      props.emailAvailability(e.target.value);
+    }
+  };
+  const emailhandlerChange = (e) => {
+    props.emailAvailabilityReset({ email: e.target.value });
+
+  };
+  const passwordhandlerBlur = (e) => {
+
+    if (e.target.value !== "") {
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/component/index.js]: use util function.

File: src/components/full-calender/components/shared/register-form/component/index.js [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+    const {registerFormEmail: {value: email}, registerFormPassword: {value: password}, registerFormUsername: {value: username}} = props;
+
+    props.registerUserDetails({username, password, email});
+  };
+
+  const validateForm = (props) => {
+
+    const {registerFormEmail: {error: emailError}, registerFormPassword: {error: passwordError}, registerFormUsername: {error: usernameError}} = props;
+
+    let flg = [!emailError, !passwordError, !usernameError].every(a => a);
```
Comment (sivasankar9) [inline src/components/full-calender/components/shared/register-form/component/index.js]: update flg to const isButtonEnable.

**Review by sivasankar9 — CHANGES_REQUESTED** [review]

**Review by sivasankar9 — APPROVED** [review]

**Pull request merged**